### PR TITLE
libzip: add up to v1.10.1

### PR DESCRIPTION
**Pull request by aumuell**
- update homepage and change download url to GitHub
- change build system to CMake for releases starting with 1.4

### CODE REVIEW

**Comment by aumuell** [comment]
@spackbot fix style

File: var/spack/repos/builtin/packages/libzip/package.py [hunk 1]
```diff
@@ -6,20 +6,42 @@
...
+    # older releases are available from the author's web site
+    version("1.5.2", sha256="be694a4abb2ffe5ec02074146757c8b56084dbcebf329123c84b205417435e15")
+    version("1.4.0", sha256="759a89690e155ca52247638b9f97e16e48fbb6077abb7ce2d921dd5b81259940")
     version("1.3.2", sha256="ab4c34eb6c3a08b678cd0f2450a6c57a13e9618b1ba34ee45d00eb5327316457")
-    version("1.2.0", sha256="6cf9840e427db96ebf3936665430bab204c9ebbd0120c326459077ed9c907d9f")
+    version(
+        "1.2.0",
+        sha256="6cf9840e427db96ebf3936665430bab204c9ebbd0120c326459077ed9c907d9f",
+        url="https://nih.at/libzip/libzip-1.2.0.tar.gz",
+    )
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/libzip/package.py]: I cannot confirm any of these sha256s using 'spack checksum'.

**Review by tldahlgren — COMMENTED** [review]
I suspect a different url argument or `url_for_version` (https://spack.readthedocs.io/en/latest/packaging_guide.html#version-urls) is needed for 1.2.0 to 1.6.1.

**Comment by aumuell** [comment]
Duh, I messed that up in multiple ways:
- the URL for 1.6.1 is special
- I broke the mechanism for extrapolating the URL based on the nearest version having a URL by adding the global `url` field

Also for me, fetching of the old versions from `libzip.org` was not possible until I added `url_fetch_method: curl` to my `config.yaml`

**Comment by aumuell** [comment]
@spackbot fix style

File: var/spack/repos/builtin/packages/libzip/package.py [hunk 2]
```diff
@@ -6,20 +6,42 @@
...
+    version("1.8.0", sha256="30ee55868c0a698d3c600492f2bea4eb62c53849bcf696d21af5eb65f3f3839e")
+    version("1.7.3", sha256="0e2276c550c5a310d4ebf3a2c3dfc43fb3b4602a072ff625842ad4f3238cb9cc")
+    version(
+        "1.6.1",
+        sha256="06eb8e9141fd19e2788cabaea9c9c2fd4d488d9e1484eb474bbfcac78e7b1d88",
+        url="https://github.com/nih-at/libzip/releases/download/rel-1-6-1/libzip-1.6.1.tar.gz",
+    )
+    # older releases are available on libzip.org
+    version("1.5.2", sha256="be694a4abb2ffe5ec02074146757c8b56084dbcebf329123c84b205417435e15")
+    version("1.4.0", sha256="759a89690e155ca52247638b9f97e16e48fbb6077abb7ce2d921dd5b81259940")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/libzip/package.py]: Are these worth adding if they can't be fetched with the default `urllib` fetcher?

File: var/spack/repos/builtin/packages/libzip/package.py [hunk 3]
```diff
@@ -6,20 +6,42 @@
...
     version("1.3.2", sha256="ab4c34eb6c3a08b678cd0f2450a6c57a13e9618b1ba34ee45d00eb5327316457")
     version("1.2.0", sha256="6cf9840e427db96ebf3936665430bab204c9ebbd0120c326459077ed9c907d9f")
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/libzip/package.py]: Perhaps these should be deprecated (https://spack.readthedocs.io/en/latest/packaging_guide.html#deprecating-old-versions) due to the fetcher issue?

**Review by tldahlgren — COMMENTED** [review]
Suggest deprecating the older versions.

File: var/spack/repos/builtin/packages/libzip/package.py (hunk 2 above)
Comment (aumuell) [reply var/spack/repos/builtin/packages/libzip/package.py]: fine for me to not have those old versions

File: var/spack/repos/builtin/packages/libzip/package.py (hunk 3 above)
Comment (aumuell) [reply var/spack/repos/builtin/packages/libzip/package.py]: I followed your suggestion here as well. Thank you!

**Review by tldahlgren — APPROVED** [review]

**Pull request merged**